File: app/CausalityCSharp/CausalityLibrary/Type/CustomAction.cs

namespace CausalityLibrary.Type
{
    public sealed class CustomAction : Action
    {
        public string CustomValue { get; set; }

        public CustomAction(SerialNumber serialNumber, string description,
    string actorName, string defaultValue = null)
            : base(serialNumber, description, actorName, defaultValue)
        {
        }

        // Don't instantiate with default constructor!
        private CustomAction()
        { }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/CustomAction.java
package com.inha.dsp.causality.type;

import java.io.Serializable;

public class CustomAction extends Action implements Serializable {
    private String customValue;
    public String getCustomValue() {
        return customValue;
    }
    public void setCustomValue(String cv) {
        customValue = cv;
    }

    public CustomAction(SerialNumber sn, String desc, String aname,
        String dvalue)
    {
        super(sn, desc, aname, dvalue);
    }

    // Don't instantiate with default constructor!
    CustomAction() {}
}
